tracer/src/Datadog.Trace/SpanContextPropagator.cs

```diff
@@ -159,12 +159,22 @@ public SpanContext Extract<T>(T carrier, Func<T, string, IEnumerable<string>> ge
             return new SpanContext(traceId, parentId, samplingPriority, null, origin);
         }
 
-        public IEnumerable<KeyValuePair<string, string>> ExtractHeaderTags<T>(T headers, IEnumerable<KeyValuePair<string, string>> headerToTagMap, string defaultTagPrefix)
+        public IEnumerable<KeyValuePair<string, string>> ExtractHeaderTags<T>(T headers, IEnumerable<KeyValuePair<string, string>> headerToTagMap, string defaultTagPrefix, string useragent = null)
             where T : IHeadersCollection
         {
             foreach (KeyValuePair<string, string> headerNameToTagName in headerToTagMap)
             {
-                string headerValue = ParseString(headers, headerNameToTagName.Key);
+                string headerValue;
+                if (headerNameToTagName.Key == HttpHeaderNames.UserAgent && !string.IsNullOrEmpty(useragent))
```

Comment:
Should this be ignore case?
```suggestion
                if (string.Equals(headerNameToTagName.Key, HttpHeaderNames.UserAgent, StringComparison.OrdinalIgnoreCase) && !string.IsNullOrEmpty(useragent))
```